airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/BaseS3Destination.kt

```diff
@@ -21,7 +21,9 @@ private val LOGGER = KotlinLogging.logger {}
 abstract class BaseS3Destination
 protected constructor(
     protected val configFactory: S3DestinationConfigFactory = S3DestinationConfigFactory(),
-    protected val environment: Map<String, String> = System.getenv()
+    protected val environment: Map<String, String> = System.getenv(),
+    private val memoryRatio: Double = 0.5,
+    private val nThreads: Int = 5
```

Comment:
Should this be some function of the number of CPUs that are allocated to the container?

Reply:
Probably, but it's hardcoded to five currently. Also according to the [resource runbook](https://airbyte.retool.com/apps/9f36acc6-644f-11ed-b1fa-0b31d64a95ed/Connection%20Resource%20Management?_releaseVersion=latest), the syncs only have 1 core by default? I assumed the point was to parallelize workloads that we assumed would be network bound (but because we're staging everything to a local file before syncing, that's probably less true?)